File: app/src/main/java/me/wcy/ponymusic/utils/JsonCallback.java
package me.wcy.ponymusic.utils;

import com.alibaba.fastjson.JSON;
import com.squareup.okhttp.Response;
import com.zhy.http.okhttp.callback.Callback;

import java.io.IOException;

/**
 * Json封装
 * Created by wcy on 2015/12/20.
 */
public abstract class JsonCallback<T> extends Callback<T> {
    private Class<T> mClass;

    public JsonCallback(Class<T> clazz) {
        this.mClass = clazz;
    }

    @Override
    public T parseNetworkResponse(Response response) throws IOException {
        String jsonString = response.body().string();
        return JSON.parseObject(jsonString, mClass);
    }
}

File: app/src/main/java/me/wcy/music/http/JsonCallback.kt
package me.wcy.music.http;

import com.google.gson.Gson;
import com.zhy.http.okhttp.callback.Callback;

import java.io.IOException;

import okhttp3.Response;

/**
 * Json封装
 * Created by wcy on 2015/12/20.
 */
public abstract class JsonCallback<T> extends Callback<T> {
    private Class<T> clazz;
    private Gson gson;

    public JsonCallback(Class<T> clazz) {
        this.clazz = clazz;
        gson = new Gson();
    }

    @Override
    public T parseNetworkResponse(Response response, int id) throws IOException {
        try {
            String jsonString = response.body().string();
            return gson.fromJson(jsonString, clazz);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }
}
